quartz-core/src/main/java/org/quartz/core/QuartzSchedulerThread.java

```diff
@@ -276,6 +277,11 @@ public void run() {
                 }
 
                 int availThreadCount = qsRsrcs.getThreadPool().blockForAvailableThreads();
+                synchronized (sigLock) {
```

Comment:
**_halted_** is an atomic variable, is synchronization required to fetch it's value?

Reply:
though synchronization is not required, but no harm by keeping it

Reply:
> **_halted_** is an atomic variable, is synchronization required to fetch it's value?

I agree - I did it to remain consistent with existing code.   I think that long ago this may have not been an AtomicBoolean - because the code even pre-dates java 1.5 (when Atomic types were introduced) , perhaps it wasn't fully cleaned up when switching to atomic, or perhaps it is still purposely trying to align with all the other points that sync up on scheduling changes.    

So to repeat: Overall I believe it unnecessary, but again, for consistency...